pkg/agent/conversation.go

```diff
@@ -171,7 +171,6 @@ func (a *Conversation) RunOneRound(ctx context.Context, query string) error {
 		}
```

Comment:
Another idea in response to most recent changes (making tool-use-shim-aware), we can probably have fnResultToShimFunctionResult at the end and ignore tool-id etc. if `EnableToolUseShim`, that way atleast process part of the response will be agnostic to the tool-use-shim or not.